index.js

```diff
@@ -1,4 +1,6 @@
 const { extendConfig } = require('hardhat/config');
+const { HardhatPluginError } = require('hardhat/plugins');
+const { name: PLUGIN_NAME } = require('./package.json')
```

Comment:
Importing the package name because hardhat actually deprecated the usage of `HardhatPluginError` without the plugin name as the first argument.